<?xml version="1.0"?>
<launch>
  <!-- Arguments -->
  <arg name="cmd_vel_topic" default="taurob_tracker/cmd_vel_raw" />
  <arg name="odom_topic" default="odom" />
  <arg name="move_forward_only" default="false"/>

  <!-- move_base -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <rosparam file="$(find taurob_mapping)/params/base_local_planner_params.yaml" command="load" />
    <rosparam file="$(find taurob_mapping)/params/costmap_common_params.yaml" command="load" ns="global_costmap" /> 
    <rosparam file="$(find taurob_mapping)/params/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find taurob_mapping)/params/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find taurob_mapping)/params/global_costmap_params.yaml" command="load" /> 
	
    
    <rosparam file="$(find turtlebot3_navigation)/param/move_base_params.yaml" command="load" />
    <rosparam file="$(find turtlebot3_navigation)/param/dwa_local_planner_params_$(arg model).yaml" command="load" />
    

    <remap from="cmd_vel" to="$(arg cmd_vel_topic)"/>
    <remap from="odom" to="$(arg odom_topic)"/>
    <param name="DWAPlannerROS/min_vel_x" value="0.0" if="$(arg move_forward_only)" />
  </node>

</launch>
